### PR TITLE
feat: Add "AutoApproveDestroy" mode option

### PR DESCRIPTION
**Pull request by bmitchinson**
## Changes

- Adds "AutoApproveDestroy" as an option.
  - Execute callbacks are optional as a result
- Updates docs
  - Verbiage: Removes "synthetization" in favor of "synthesis"
  - Adds info re "AutoApproveDestroy"

## Details

> _...Why would you want an action to destroy infra?_

I'm using this action for manually triggered workflows only, not for PR usage.

I have a password protected workflow for someone to deploy demo purpose infra (auto approve) and destroy (pending this pr, auto destroy). 

This new option isn't intended to ever be used on PRs, so there's no PR comment callbacks after executing the `cdktf destroy`.

Thanks for review. While manual workflows is a bit of an edge case, I don't think it makes the primary usage for auto applying and yielding PR feedback any more confusing or abstracted. With that in mind, would love to see this added!

❤️

### CODE REVIEW

File: src/action.ts [hunk 1]
```diff
@@ -66,12 +66,13 @@ enum ExecutionMode {
...
   PlanOnly = "plan-only",
   AutoApproveApply = "auto-approve-apply",
+  AutoApproveDestroy = "auto-approve-destroy",
 }
 
 async function execute(
   cdktfCommand: string,
-  reportSuccess: (output: string, runUrl?: string) => Promise<void>,
-  reportFailure: (
+  reportSuccess?: (output: string, runUrl?: string) => Promise<void>,
```
Comment (DanielMSchmidt) [inline src/action.ts:74]: Let's keep these required and report back to the user. One can still disable it by setting commentOnPR to false

**Review by DanielMSchmidt — CHANGES_REQUESTED** [review]
Great idea to add this, I got one change I'd like to see :)

File: src/action.ts (hunk 1 above)
Comment (bmitchinson) [reply src/action.ts:74]: When you say "report back to the user", what would you like that change to look like. 
Leaving a PR comment like the others? Or logging a message in the action with a console log?

I didn't think you'd want to encourage PRs that teardown infra, but I suppose having the option to leave a comment afterward doesn't make that any more or less likely to happy.

**Comment by bmitchinson** [comment]
Updated to leave PR comments

**Review by DanielMSchmidt — APPROVED** [review]

**Comment by DanielMSchmidt** [comment]
Could you run yarn build and commit to make the check green?

**Comment by bmitchinson** [comment]
What is the release cadence for this repo? Wondering when I can use this in the markplace action.

**Comment by DanielMSchmidt** [comment]
It's released now, normally it;'s released after merge but our automation had a problem

**Pull request merged**